add a new file ratInAMaze

nayanajain opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

nayanajain committed:
add a new file ratInAMaze

nayanajain committed:
added comments and input output

nayanajain committed:
updated readme

nayanajain commented:
updated readme file as well...please merge my code.

HarshCasper commented on C-Plus-Plus/cp/ratInAMaze.cpp:
```diff
@@ -0,0 +1,124 @@
+/*There is a rat who is at the starting point of a matrix and wants to reach
+ to the ending point i.e at the last coordinate of the matrix.
+ 1 denotes the point where it is possible for the mouse to go and
+ 0 denotes that the mouse cannot reach at that point. */
```
Can you add more details about the Problem here 😅 

Also if possible add a Problem Link on some Online Coding Platform

HarshCasper commented on C-Plus-Plus/cp/ratInAMaze.cpp:
```diff
@@ -0,0 +1,124 @@
+
+int main()
+{
+    int m, n;
```
Use meaningful variable names

HarshCasper reviewed (changes requested):
Do these changes as well: 

- Remove extra lines of code 
- Format the Code to make it look readable
- Put Comments above the Code, not beside it

nayanajain committed:
done the requested changes

nayanajain committed:
done the requested changes

nayanajain commented:
done the requested changes

HarshCasper committed:
Merge branch 'master' into nj

HarshCasper commented on C-Plus-Plus/README.md:
```diff
@@ -47,6 +47,13 @@
 - [Suduko Solver](cp/SudukoSolver.cpp)
 - [Target Sum triplets](cp/target_sum_triplets.cpp)
 - [Z Algorithm](cp/zalgorithm.cpp)
+- [ArraySub](cp/ARRAYSUB.cpp)
+- [Suduko Solver](cp/SudukoSolver.cpp)
+- [Josephus Problem](cp/Josephus.cpp)
+- [NQueens](cp/NQueens.cpp)
+- [rat in a maze](cp/ratInAMaze.cpp)
+- [Longest Subarray with Zero Sum](cp/ZeroSum.cpp)
+- [Next Largest Permutation](cp/NextPermutation.cpp)
```
Why so many README changes?

HarshCasper replied:
The Question still remains.

HarshCasper commented on C-Plus-Plus/cp/ratInAMaze.cpp:
```diff
@@ -0,0 +1,124 @@
+    return 0;
+}
+
+
```
Remove extra lines of code

HarshCasper commented on C-Plus-Plus/cp/ratInAMaze.cpp:
```diff
@@ -0,0 +1,124 @@
+    {
+        solution[x][y]=1;
+
+//mpving in right direction
```
Can you remove the typos? Also properly format the Comments to make them look meaningful

nayanajain committed:
done the changes

nayanajain committed:
done the changes..

nayanajain committed:
updated typo and formatted comments

nayanajain committed:
Merge branch 'nj' of https://github.com/nayanajain/NeoAlgo into nj

HarshCasper reviewed (changes requested):
Remove unnecessary lines of Code

nayanajain commented:
I have added a single line that is name of the file in my readme...I am not able to get why showing so many changes
I have added link of the problem and removed blank lines between the code..please merge my code

If my pr not gets merged, will I get points for issue??please clear my query
Thank you

On Thu, 21 Jan 2021, 12:03 pm Nayana Jain, <nayanajain854@gmail.com> wrote:

> I have added only a single line (rat in a maze) in the Readme file ..I am
> not able to understand  your change request
>
> On Thu, 21 Jan 2021, 7:41 am Harsh Bardhan Mishra, <
> notifications@github.com> wrote:
>
>> *@HarshCasper* requested changes on this pull request.
>>
>> Remove unnecessary lines of Code
>> ------------------------------
>>
>> In C-Plus-Plus/README.md
>> <https://github.com/TesseractCoding/NeoAlgo/pull/1882#discussion_r561474509>
>> :
>>
>> > +- [ArraySub](cp/ARRAYSUB.cpp)
>> +- [Suduko Solver](cp/SudukoSolver.cpp)
>> +- [Josephus Problem](cp/Josephus.cpp)
>> +- [NQueens](cp/NQueens.cpp)
>> +- [rat in a maze](cp/ratInAMaze.cpp)
>> +- [Longest Subarray with Zero Sum](cp/ZeroSum.cpp)
>> +- [Next Largest Permutation](cp/NextPermutation.cpp)
>>
>> The Question still remains.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/TesseractCoding/NeoAlgo/pull/1882#pullrequestreview-572929213>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AL7HKZGFCQG7JKYY2E7G22LS26ELHANCNFSM4VQZYQ3A>
>> .
>>
>

nayanajain committed:
done requested changes by adding link to the problem and removing unnecessary lines

nayanajain committed:
added file Armstrong_Number.py

nayanajain committed:
Delete Armstrong_Number.py
